### PR TITLE
vieb: Add version 4.0.0

### PR DESCRIPTION
**Pull request by run2cmd**
I'd like to add very promising browser [Vieb](https://vieb.dev/). I use that day to day and would be great to have Scoop packages for it. 

I have tested installation and update process with my own test bucket under https://github.com/run2cmd/scoop-test-bucket

### CODE REVIEW

File: bucket/vieb.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "version": "4.0.0",
+    "description": "Vieb is the Vim Inspired Electron Browser",
+    "homepage": "https://vieb.dev/",
+    "license": "GPL-3.0-only",
```
Comment (Ash258) [inline bucket/vieb.json:5]: invalid SPX identifier

File: bucket/vieb.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            "Vieb.4.0.0.exe",
+            "Vieb"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/Jelmerro/Vieb"
+    },
+    "autoupdate": {
+        "url": "https://github.com/Jelmerro/Vieb/releases/download/$version/Vieb.$version.exe",
+        "extract_dir": "vieb-$version-x64"
```
Comment (Ash258) [inline bucket/vieb.json:19]: ???

File: bucket/vieb.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    "version": "4.0.0",
+    "description": "Vieb is the Vim Inspired Electron Browser",
```
Comment (Ash258) [inline bucket/vieb.json:3]: Do not mention applicaiton name

File: bucket/vieb.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "homepage": "https://vieb.dev/",
+    "license": "GPL-3.0-only",
+    "url": "https://github.com/Jelmerro/Vieb/releases/download/4.0.0/Vieb.4.0.0.exe",
+    "hash": "d5f46c3fb5a8a379866f4a97cd2ad5622fc6430465623e291aefb0dd1f8e6e4a",
+    "shortcuts": [
+        [
+            "Vieb.4.0.0.exe",
```
Comment (Ash258) [inline bucket/vieb.json:10]: Having autoupdate and version in shortcut does not make sense and will lead to failed installation after autoupdate

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/vieb.json (hunk 1 above)
Comment (run2cmd) [reply bucket/vieb.json:5]: This is what I found on https://spdx.org/licenses/GPL-3.0-only.html. What identifier I should put there?

Comment (Ash258) [reply bucket/vieb.json:5]: Sorry my bad. I overlooked it.

File: bucket/vieb.json (hunk 4 above)
Comment (run2cmd) [reply bucket/vieb.json:10]: This is self execute binary. I'll talk with developer if we can have file without version in name.

Comment (Ash258) [reply bucket/vieb.json:10]: <url>#/view.exe

File: bucket/vieb.json (hunk 2 above)
Comment (run2cmd) [reply bucket/vieb.json]: Removed.

File: bucket/vieb.json (hunk 3 above)
Comment (run2cmd) [reply bucket/vieb.json]: Changed.

File: bucket/vieb.json (hunk 1 above)
Comment (Jelmerro) [reply bucket/vieb.json:5]: I want to mention that it should actually be https://spdx.org/licenses/GPL-3.0-or-later.html

File: bucket/vieb.json (hunk 4 above)
Comment (Jelmerro) [reply bucket/vieb.json:10]: Would it be ok if the zip is versioned, but the exe inside it isn't? For example `Vieb-4.1.0-win.zip` containing `Vieb.exe` inside at the root of the zip?

Comment (Ash258) [reply bucket/vieb.json:10]: Does not make sense to waste time with different packaging. As already said. Just use URL rewrite `<url>#/view.exe`

Comment (run2cmd) [reply bucket/vieb.json]: I have tested both the rewrite option and with zip file. Rewrite was waring however when using Vieb from single exe I had issues with creating shortcuts of my own for example on task bar. I did not experience those issues with zip file. Sine it's already created I used that one.

File: bucket/vieb.json (hunk 1 above)
Comment (run2cmd) [reply bucket/vieb.json]: Updated license to GPL-3.0-or-later

File: bucket/vieb.json (hunk 3 above)
Comment (Ash258) [reply bucket/vieb.json:3]: Again

File: bucket/vieb.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+    "version": "4.1.0",
+    "description": "Vieb is the Vim Inspired Electron Browser",
+    "homepage": "https://vieb.dev/",
+    "license": "GPL-3.0-or-later",
+    "hash": "1c958e4cba47248213f4b9ede909ea6f57cf24967e953fceb489b5342b1f6f88",
```
Comment (Ash258) [inline bucket/vieb.json:6]: Hash is checked after the URL is downloaded. Does not make sense to have the hash above URL

File: bucket/vieb.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "url": "https://github.com/Jelmerro/Vieb/releases/download/4.1.0/Vieb-4.1.0-win.zip",
```
Comment (Ash258) [inline bucket/vieb.json:7]: This has to be 64bit version

File: bucket/vieb.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        [
+            "Vieb.exe",
+            "Vieb"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/Jelmerro/Vieb"
+    },
+    "autoupdate": {
+        "url": "https://github.com/Jelmerro/Vieb/releases/download/$version/Vieb-$version-win.zip"
```
Comment (Ash258) [inline bucket/vieb.json:19]: 64bit

File: bucket/vieb.json (hunk 6 above)
Comment (Ash258) [inline bucket/vieb.json:7]: What does portable means? Where are application data and configurations stored?

image

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/vieb.json (hunk 5 above)
Comment (run2cmd) [reply bucket/vieb.json]: Fixed

File: bucket/vieb.json (hunk 6 above)
Comment (run2cmd) [reply bucket/vieb.json]: Added.

File: bucket/vieb.json (hunk 7 above)
Comment (run2cmd) [reply bucket/vieb.json]: Added.

File: bucket/vieb.json (hunk 6 above)
Comment (run2cmd) [reply bucket/vieb.json]: The initial configuration is not needed. Vieb reads `%APPDATA%\Vieb\viebrc` file if its created.

File: bucket/vieb.json (hunk 3 above)
Comment (run2cmd) [reply bucket/vieb.json]: Fixed

**Comment by run2cmd** [comment]
@Ash258 Anything else you'd like me to improve about this PR? Do you want me to squash changes, the commit history in this PR is a bit messy.

**Review by Ash258 — APPROVED** [review]

**Pull request merged**